zrml/neo-swaps/src/migration.rs

```diff
@@ -15,6 +15,239 @@
 // You should have received a copy of the GNU General Public License
 // along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
 
+use crate::{
+    traits::LiquiditySharesManager, types::Pool, AssetOf, BalanceOf, Config, LiquidityTreeOf,
+    Pallet, Pools,
+};
+use alloc::collections::BTreeMap;
+use core::marker::PhantomData;
+use frame_support::{
+    dispatch::Weight,
+    log,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+    RuntimeDebug,
+};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::Saturating;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "try-runtime")] {
+        use crate::{MarketIdOf};
+        use alloc::{format, vec::Vec};
+        use frame_support::{migration::storage_key_iter, pallet_prelude::Twox64Concat};
+    }
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(any(feature = "try-runtime", test))] {
+        const NEO_SWAPS: &[u8] = b"NeoSwaps";
+        const POOLS: &[u8] = b"Pools";
+    }
+}
+
+const NEO_SWAPS_REQUIRED_STORAGE_VERSION: u16 = 1;
+const NEO_SWAPS_NEXT_STORAGE_VERSION: u16 = NEO_SWAPS_REQUIRED_STORAGE_VERSION + 1;
+
+#[derive(Clone, Decode, Encode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct OldPool<T, LSM>
+where
+    T: Config,
+    LSM: LiquiditySharesManager<T>,
+{
+    pub account_id: T::AccountId,
+    pub reserves: BTreeMap<AssetOf<T>, BalanceOf<T>>,
+    pub collateral: AssetOf<T>,
+    pub liquidity_parameter: BalanceOf<T>,
+    pub liquidity_shares_manager: LSM,
+    pub swap_fee: BalanceOf<T>,
+}
```

Comment:
`BTreeMap` to `BoundedBTreeMap` does not require any migration because the underlying scale encoded data remains the same. The new generic parameter that specifies the boundary is only used on the logical layer, e.g. to ensure that nothing is added beyond bounds or decoding fails if boundaries are surpassed, but the data layout remains the same.

https://docs.rs/bounded-collections/0.2.0/src/bounded_collections/bounded_btree_map.rs.html#34

Reply:
Other than that, migration looks good.

Reply:
Hmm, know I feel silly. I think in that case it makes more sense to just remove the entire migration.

Reply:
I'm fine with either way. But it seems you have some additional useful sanity checks that already seem to make it worth executing it https://github.com/zeitgeistpm/zeitgeist/blob/8415c118c8fe564683f4868fb04f77def3d84f35/zrml/neo-swaps/src/migration.rs#L90-L93